### PR TITLE
Implementing Schmidt 2010 Structure Function calculation method

### PR DESCRIPTION
**Pull request by drewoldag**
This PR represents the work to implement the Schmidt 2010 structure function calculator method. Defined here as Eqn. 2: https://iopscience.iop.org/article/10.1088/0004-637X/714/2/1194/pdf

This PR is similar is scope and form to:
- PR #106 
- PR #100

### CODE REVIEW

File: tests/lsstseries_tests/structure_function_calculators/test_schmidt_2010_calculator.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    arg_container.combine = False
+
+    sf_calculator = Schmidt2010StructureFunctionCalculator(test_t, test_y, test_e, arg_container)
+
+    res = sf_calculator.calculate()
+
+    assert res
+
+
+def test_sf2_base_case_bauer_2009b():
```
Comment (delucchi-cmu) [inline tests/lsstseries_tests/structure_function_calculators/test_schmidt_2010_calculator.py:27]:
```suggestion
def test_sf2_base_case_schmidt_2010():
```

File: tests/lsstseries_tests/structure_function_calculators/test_schmidt_2010_calculator.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    """
+    Base test case accessing calc_sf2 directly. Uses `Bauer 2009b` SF calculation method.
```
Comment (delucchi-cmu) [inline tests/lsstseries_tests/structure_function_calculators/test_schmidt_2010_calculator.py:29]:
```suggestion
    Base test case accessing calc_sf2 directly. Uses `Schmidt 2010` SF calculation method.
```

File: src/lsstseries/analysis/structure_function/schmidt_2010/calculator.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+from lsstseries.analysis.structure_function.base_calculator import StructureFunctionCalculator
+
+SQRT_PI_OVER_2 = np.sqrt(np.pi / 2.0)
+
+
+class Schmidt2010StructureFunctionCalculator(StructureFunctionCalculator):
+    """This class implements the structure function calculation described in
+    Eqn. 2 of Schmidt et al. 2010, 2010ApJ...714.1194S [https://arxiv.org/abs/1002.2642]
+
+    `V(delta_t) = mean(sqrt(pi/2) * abs(delta_flux_i,j) - sqrt(err_i^2 + err_j^2))`
```
[the remark this answers is not in the document]
Comment (nevencaplar) [reply src/lsstseries/analysis/structure_function/schmidt_2010/calculator.py:13]: The quantity V is not variance, it is a square root of variance, i.e., structure function. Keep it as SF - all the calculators should have SF as what they are calculating. I.e., all docstrings should have an equation that starts with SF

[the remark this answers is not in the document]
Comment (nevencaplar) [reply src/lsstseries/analysis/structure_function/schmidt_2010/calculator.py:13]: This paper also has an erratum: https://iopscience.iop.org/article/10.1088/0004-637X/721/2/1941. So, change V to SF, remove or update the explanation, cite erratum as well in the docstring

[the remark this answers is not in the document]
Comment (drewoldag) [reply src/lsstseries/analysis/structure_function/schmidt_2010/calculator.py:13]: Gotcah, I can do that.

**Review by nevencaplar — APPROVED** [review]
Looks good to me

**Pull request merged**